File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    n = len(goal)
    for i in range(n):
        if len(cards1) and cards1[0] == goal[0]:
            cards1.pop(0)
            goal.pop(0)
        elif len(cards2) and cards2[0] == goal[0]:
            cards2.pop(0)
            goal.pop(0)
    
    if not len(goal):
        return "Yes"
    else:
        return "No"
    

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    cards1 = cards1.reverse();
    cards2 = cards2.reverse();
    goal = goal.reverse();
    
    while (goal.length) {
        if (cards1.length && cards1[cards1.length - 1] === goal[goal.length - 1]) {
            cards1.pop();
            goal.pop();
        } else if (cards2.length && cards2[cards2.length - 1] === goal[goal.length - 1]) {
            cards2.pop();
            goal.pop();
        } else {
            return "No";
        }
    }
    return "Yes";
}